prow/gerrit/adapter/adapter.go

```diff
@@ -435,21 +449,26 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		messages := currentMessages(change, lastUpdate)
 		logger.WithField("failed", len(failed)).Debug("Failed jobs parsed from previous comments.")
 		filters := []pjutil.Filter{
-			messageFilter(messages, failed, all, logger),
+			messageFilter(messages, failed, all, triggerTimes, logger),
 		}
 		// Automatically trigger the Prow jobs if the revision is new and the
 		// change is not in WorkInProgress.
 		if revision.Created.Time.After(lastUpdate) && !change.WorkInProgress {
-			filters = append(filters, pjutil.NewTestAllFilter())
+			filters = append(filters, &timeAnnotationFilter{
+				Filter:       pjutil.NewTestAllFilter(),
+				eventTime:    revision.Created.Time,
+				triggerTimes: triggerTimes,
+			})
 		}
 		toTrigger, err := pjutil.FilterPresubmits(pjutil.NewAggregateFilter(filters), listChangedFiles(change), change.Branch, presubmits, logger)
 		if err != nil {
 			return fmt.Errorf("filter presubmits: %w", err)
 		}
+		// At this point triggerTimes should be properly populated as a side effect of FilterPresubmits.
 
 		// Reply with help information to run the presubmit Prow jobs if requested.
 		for _, msg := range messages {
-			needsHelp, note := pjutil.ShouldRespondWithHelp(msg, len(toTrigger))
+			needsHelp, note := pjutil.ShouldRespondWithHelp(msg.Message, len(toTrigger))
```

Comment:
should we also report the duration here? Like from the time a comment ask for help to the time prow comment back?

Reply:
I didn't include that originally since it isn't a job being triggered, but I guess that makes sense to include if we're actually talking about processing latency. It shouldn't account for a significant amount of the metrics anyways.
Added.